### PR TITLE
Insert new MNIST example program from the Chinese README.md into the English version

### PR DESCRIPTION
**Pull request by wangkuiyi**
This is based on @luotao1 's work in https://github.com/PaddlePaddle/book/pull/106.

### CODE REVIEW

File: recognize_digits/README.en.md [hunk 1]
```diff
@@ -132,283 +125,160 @@ Execute the following command to download the [MNIST](http://yann.lecun.com/exdb
...
-    inputs(img, lbl)
-    outputs(classification_cost(input=predict, label=lbl))
-else:
-    outputs(predict)
-```
-
-#### Softmax Regression
-
-One simple fully connected layer with softmax activation function outputs classification result.
+- softmax regression: the network has a fully-connection layer and a softmax output layer:
```
Comment (helinwang) [inline recognize_digits/README.en.md]: and a softmax output layer -> with softmax activation

File: recognize_digits/README.en.md [hunk 2]
```diff
@@ -132,283 +125,160 @@ Execute the following command to download the [MNIST](http://yann.lecun.com/exdb
...
+    predict = paddle.layer.fc(input=img,
+                              size=10,
+                              act=paddle.activation.Softmax())
     return predict
 ```
 
-#### MultiLayer Perceptron
-
-The following code implements a Multilayer Perceptron with two fully connected hidden layers and a ReLU activation function. The output layer has a Softmax activation function.
+- multi-layer perceptron: this network has two hidden fully-connected layers, both using ReLU activation.  The output layer is softmax:
```
Comment (helinwang) [inline recognize_digits/README.en.md]: The output layer is softmax -> The output activation is softmax

File: recognize_digits/README.en.md [hunk 3]
```diff
@@ -132,283 +125,160 @@ Execute the following command to download the [MNIST](http://yann.lecun.com/exdb
...
+    predict = paddle.layer.fc(input=hidden2,
+                              size=10,
+                              act=paddle.activation.Softmax())
     return predict
 ```
 
-#### Convolutional Neural Network LeNet-5
-
-The following is the LeNet-5 network architecture. A 2D input image is first fed into two sets of convolutional layers and pooling layers, this result is then fed to a fully connected layer, and another fully connected layer with a softmax activation.
+- convolution network LeNet-5: the input image is fed through two convolution-pooling layer, a fully-connected layer, and the softmax output layer:
```
Comment (helinwang) [inline recognize_digits/README.en.md:163]: a fully-connected layer, and the softmax output layer -> a fully-connected layer,with softmax activation

File: recognize_digits/README.en.md [hunk 4]
```diff
@@ -132,283 +125,160 @@ Execute the following command to download the [MNIST](http://yann.lecun.com/exdb
...
-
-python -m paddle.utils.plotcurve -i $log > plot.png
+    predict = paddle.layer.fc(input=fc1,
+                              size=10,
+                              act=paddle.activation.Softmax())
+    return predict
 ```
 
-After configuring parameters, execute `./train.sh`. Training log is as follows.
+PaddlePaddle provides a special layer `layer.data` for reading data. Let us create a data layer for reading images and connect it to a classification network created using one of above three functions.  We also need a cost layer for training the model.
```
Comment (helinwang) [inline recognize_digits/README.en.md:196]: `.  We also` -> `. We also`

File: recognize_digits/README.en.md [hunk 5]
```diff
@@ -132,283 +125,160 @@ Execute the following command to download the [MNIST](http://yann.lecun.com/exdb
...
-```text
-Best pass is 00013, testing Avgcost is 0.484447
-The classification accuracy is 90.01%
+    trainer = paddle.trainer.SGD(cost=cost,
+                                 parameters=parameters,
+                                 update_equation=optimizer)
 ```
 
-From the evaluation results, the best pass for softmax regression model is pass-00013, where the classification accuracy is 90.01%, and the last pass-00099 has an accuracy of 89.3%. From Fig. 7, we also see that the best accuracy may not appear in the last pass. This is because during training, the model may already arrive at a local optimum, and it just swings around nearby in the following passes, or it gets a lower local optimum.
+Then we specify the training data `paddle.dataset.movielens.train()` and testing data `paddle.dataset.movielens.test()`.  These two functions are *reader creators*, once called, returns a *reader*.  A reader is a Python function, which, once called, returns a Python yield generator, which yields instances of data.
```
Comment (helinwang) [inline recognize_digits/README.en.md]: Python yield generator -> Python generator

File: recognize_digits/README.en.md [hunk 6]
```diff
@@ -132,283 +125,160 @@ Execute the following command to download the [MNIST](http://yann.lecun.com/exdb
...
+                                 update_equation=optimizer)
 ```
 
-From the evaluation results, the best pass for softmax regression model is pass-00013, where the classification accuracy is 90.01%, and the last pass-00099 has an accuracy of 89.3%. From Fig. 7, we also see that the best accuracy may not appear in the last pass. This is because during training, the model may already arrive at a local optimum, and it just swings around nearby in the following passes, or it gets a lower local optimum.
+Then we specify the training data `paddle.dataset.movielens.train()` and testing data `paddle.dataset.movielens.test()`.  These two functions are *reader creators*, once called, returns a *reader*.  A reader is a Python function, which, once called, returns a Python yield generator, which yields instances of data.  
 
-### Results of Multilayer Perceptron
+Here `shuffle` is a reader decorator, which takes a reader A as its parameter, and returns a new reader B, where B calls A to read in `buffer_size` data instances everytime into a buffer, then shuffles and yield instances in the buffer.  If you want very shuffled data, try use a larger buffer size.
```
Comment (helinwang) [inline recognize_digits/README.en.md]: `  If you` -> ` If you`

File: recognize_digits/README.en.md [hunk 7]
```diff
@@ -132,283 +125,160 @@ Execute the following command to download the [MNIST](http://yann.lecun.com/exdb
...
+    # Pass 0, Batch 400, Cost 0.410101, {'classification_error_evaluator': 0.125}
+    # Test with Pass 0, Cost 0.326659, {'classification_error_evaluator': 0.09470000118017197}
 ```
 
-- -c sets model architecture
-- -d sets data for prediction
-- -m sets model parameters, here the best trained model is used for prediction
-
-Follow the instructions to input image ID for prediction. The classifier can output probabilities for each digit, predictions with the highest probability, and ground truth label.
+After the training, we can choose the best model:
```
Comment (helinwang) [inline recognize_digits/README.en.md]: "we can choose the best model:"，好像只能看一看，并不能选择吧（模型没有存）。

File: recognize_digits/README.md [hunk 8]
```diff
@@ -267,31 +271,31 @@ def main():
     # Test with Pass 0, Cost 0.326659, {'classification_error_evaluator': 0.09470000118017197}
 ```
 
-最后，选出最佳模型，并评估其效果。
+训练之后，我们可以选出最佳模型，并评估其效果。
```
Comment (helinwang) [inline recognize_digits/README.md]: "选出最佳模型"，其实只能看看，选不了（没存模型）。

**Review by helinwang — APPROVED** [review]

File: recognize_digits/README.en.md (hunk 1 above)
Comment (wangkuiyi) [reply recognize_digits/README.en.md]: done

File: recognize_digits/README.en.md (hunk 2 above)
Comment (wangkuiyi) [reply recognize_digits/README.en.md]: done

File: recognize_digits/README.en.md (hunk 3 above)
Comment (wangkuiyi) [reply recognize_digits/README.en.md:163]: done

File: recognize_digits/README.en.md (hunk 4 above)
Comment (wangkuiyi) [reply recognize_digits/README.en.md:196]: done

File: recognize_digits/README.en.md (hunk 5 above)
Comment (wangkuiyi) [reply recognize_digits/README.en.md]: done

File: recognize_digits/README.en.md (hunk 6 above)
Comment (wangkuiyi) [reply recognize_digits/README.en.md]: done.

File: recognize_digits/README.en.md (hunk 7 above)
Comment (wangkuiyi) [reply recognize_digits/README.en.md]: changed. done

File: recognize_digits/README.md (hunk 8 above)
Comment (wangkuiyi) [reply recognize_digits/README.md]: changed. done.
